### PR TITLE
DOC: explain deprecation 

### PR DESCRIPTION
**Pull request by kingjr**
@Eric89GXL

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Something went terribly wrong with the commit history

**Comment by alexandrebarachant** [comment]
yep. looks like you created the new branch from where we left the spoc branch at the codding sprint.

**Comment by kingjr** [comment]
Yes, I'm on it

**Comment by kingjr** [comment]
@Eric89GXL can we do python code blocks?

**Comment by Eric89GXL** [comment]
Yep, just doing `::` at the end of a line will render the following blocked
text as code

File: doc/contributing.rst [hunk 1]
```diff
@@ -137,6 +137,30 @@ with:
...
+    def my_function():
+       return 'foo'
+
+If you need to deprecate a parameter, use the mne warning function.
+
+.. code-block:: python
+
+    from mne.utils import warn
+
+    def my_function(my_param=None):
```
Comment (agramfort) [inline doc/contributing.rst]:
```
if my_param is not None:
    warn(...)
```

**Comment by agramfort** [comment]
@kingjr I pushed the fix I requested.

merge if you're happy

File: doc/contributing.rst [hunk 2]
```diff
@@ -150,14 +150,19 @@ If you need to deprecate a function or a class, use the ``@deprecated`` decorato
...
 
 .. code-block:: python
 
     from mne.utils import warn
 
-    def my_function(my_param=None):
-       warn('my_param will be deprecated in 0.XX.', DeprecationWarning)
-       return 'foo'
+    def my_function(new_param, old_param=None):
+        if my_param is not None:
```
Comment (Eric89GXL) [inline doc/contributing.rst]: this code is broken :(

Also while you're at it feel free to make my suggested `::` change instead of using `code-block`

**Comment by agramfort** [comment]
oups... like this?

**Comment by Eric89GXL** [comment]
Yep. See it renders just fine:

https://4544-1301584-gh.circle-artifacts.com/0/home/ubuntu/mne-python/doc/_build/html/contributing.html#deprecating

**Comment by agramfort** [comment]
beautiful !

**Pull request merged**